### ISSUE TITLE
Ubuntu Linux ARM blacs link issue

### ISSUE THREAD

**Issue by scivision** [gemini3d/gemini3d#73]
I was able to replicate @mattzett issue. Matt is using Mac M1 with Ubuntu 20.04 in Parallels VM.

The error was like: 

<img width="1068" alt="Screen Shot 2021-09-30 at 9 15 29 AM" src="https://user-images.githubusercontent.com/10931741/135671292-8e1f9fde-edd1-44f3-a218-a9cf151d9b38.png">

I used Ubuntu 21.04 on Raspberry Pi 4, installed 64-bit that is

```sh
$ uname -m
aarch64
```

speed up build to get to the error like:

```sh
cmake --build build --target gemini.bin
````

---

Working on aarch64 Raspi 4 with Ubuntu 21.04:

* all system libraries e.g. `libmumps-dev libscalapack-openmpi-dev` and so on.
* `cmake -Dmumps_external=on`

NOT working on aarch64 Raspi 4:

* autobuild `cmake --preset build` all libraries (except MPI)
* `cmake -B build -Dscalapack_external=on`

---

works on CentOS 8 x86_64:

* `cmake -B build` all system libraries
* `cmake -B build -Dscalapack_external=yes`
* `cmake -B build -Dscalapack_external=yes -Dmumps_external=yes`

---

I notice on Windows MSYS2 that MPI is linked after mumps but before scalapack, yet it links fine. Is this the same link order seen on Ubuntu that fails?

**Comment by mattzett** [comment]
actually I'm seeing this on intel ubuntu 20.04 too

**Comment by scivision** [comment]
The error is due to link ordering. Some linkers are more forgiving than others about library order. On a failing system (WSL Ubuntu 20.04) gemini.bin link fails with:

```
/usr/bin/gfortran -pthread  src/CMakeFiles/gemini.bin.dir/gemini_main.f90.o -o gemini.bin  -Wl,-rpath,/usr/lib/x86_64-linux-gnu/hdf5/serial:/usr/lib/x86_64-linux-gnu/openmpi/lib:/usr/lib/x86_64-linux-gnu/lapack:/usr/lib/x86_64-linux-gnu/blas:  libgemini3d.a  src/utils/hwloc/libhwloc_ifc.a  src/utils/hwloc/libhwloc_c.a  lib/libglow.a  lib/libmsis00.a  src/vendor/nrl_msis/libmsis00mod.a  src/vendor/nrl_hwm/libhwm_ifc.a  src/utils/hwloc/libhwloc_ifc.a  src/utils/hwloc/libhwloc_c.a  /usr/lib/x86_64-linux-gnu/libhwloc.so  lib/libh5fortran.a  /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5hl_fortran.so  /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5_fortran.so  /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5_hl.so  /usr/lib/x86_64-linux-gnu/hdf5/serial/libhdf5.so  /usr/lib/x86_64-linux-gnu/libz.so  /usr/lib/x86_64-linux-gnu/libsz.so  -ldl  -lm  libnc4fortran.a  lib/libdmumps.a  lib/libmumps_common.a  lib/libpord.a  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_usempif08.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_usempi_ignore_tkr.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi_mpifh.so  /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so  lib/libscalapack.a  lib/libblacs.a  /usr/lib/x86_64-linux-gnu/lapack/liblapack.so  /usr/lib/x86_64-linux-gnu/blas/libblas.so  -lpthread && :
/usr/bin/ld: lib/libblacs.a(BI_Rsend.c.o): undefined reference to symbol 'MPI_Rsend'
/usr/bin/ld: /usr/lib/x86_64-linux-gnu/openmpi/lib/libmpi.so: error adding symbols: DSO missing from command line
```

**Comment by scivision** [comment]
fixed by a8d5d8e5952f967823399a0253176ebf25f5c3bd
